### PR TITLE
Allow choosing a custom default branch

### PR DESCRIPTION
**Pull request by alejandrohdezma**
> ❗ This PR needs to be updated with a new Scala Steward version after scala-steward-org/scala-steward#2183 is merged. It shouldn't be merged until that is done.

# What has been done in this PR?

By default, Scala Steward uses the repository's default branch to make the updates. After this PR gets merged, if you want to customize that behaviour, you can use the new `branch` input:

```yml
- name: Launch Scala Steward
  uses: scala-steward-org/scala-steward-action@v2
  with:
    github-token: ${{ github.token }}
    branch: develop
```

This input is specially useful when you want to update multiple branches. For this, you can take advantage of [GitHub Action's `matrix`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix).

```yml
# This workflow will launch at 00:00 every Sunday
on:
  schedule:
    - cron: '0 0 * * 0'
name: Launch Scala Steward on `main`, `1.x` and `2.x`
jobs:
  scala-steward:
    runs-on: ubuntu-latest
    name: Launch Scala Steward
    strategy:
      matrix:
        branch: ['main', '1.x', '2.x']
    steps:
      - name: Launch Scala Steward
        uses: scala-steward-org/scala-steward-action@v2
        with:
          github-token: ${{ secrets.REPO_GITHUB_TOKEN }}
          branch: ${{ matrix.branch }}
```

### CODE REVIEW

**Comment by alejandrohdezma** [comment]
Hey @exoego, for this PR to be merged, and the branch-feature by widely available, a new Scala Steward version needs to be published with the change, so we can update the `action.yml` in here.

File: README.md [hunk 1]
```diff
@@ -232,9 +233,46 @@ Then you can use an action like [tibdex/github-app-token](https://github.com/tib
...
+        uses: scala-steward-org/scala-steward-action@v2
+        with:
+          github-token: ${{ secrets.REPO_GITHUB_TOKEN }}
+          branch: ${{ matrix.branch }}
+```
+
 ### GPG
 
-If you want commits created by Scala Steward to be automatically signed with a GPG key, follow this steps:
+If you want commits created by Scala Steward to be automatically signed with a GPG key, follow thess steps:
```
Comment (exoego) [inline README.md:275]: TYPO
```suggestion
If you want commits created by Scala Steward to be automatically signed with a GPG key, follow these steps:
```

**Review by exoego — APPROVED** [review]
Thanks for this follow up !!
This definitely help people, specially OSS maintainers who maintain several version branches like `1.x`, `2.x` and `main`.

I am not authorized on this repo, so I just left an "approval".

**Comment by alejandrohdezma** [comment]
No problemo :) Although I don't have enough permissions here to add you as a collaborator. @fthomas in case you saw this, could you give @exoego and me permissions on this repo? Thank you very much 😸

**Comment by alejandrohdezma** [comment]
@exoego I'll wait to push a new version of the GitHub Action until a new Scala Steward version is ready. I'll send a new PR once that is done 😄

**Comment by fthomas** [comment]
@alejandrohdezma I'd like to manage access to repos of this org via teams. I'm going to add a gh-action team to the org, add you and @exoego to it and give this team `Maintainer` permissions in this repo. Does that sound OK?

**Comment by exoego** [comment]
@fthomas Sounds good to me!!

**Comment by fthomas** [comment]
Done! Ping me if you need anything else.

**Comment by alejandrohdezma** [comment]
Thanks @fthomas! :)

**Comment by fthomas** [comment]
You're welcome! @alejandrohdezma I also gave you maintainer permissions on this team so that you can add and remove members yourself.

**Pull request merged**